test/fpm_test/test_filesystem.f90

```diff
@@ -173,7 +174,86 @@ subroutine check_rmdir(error, path)
                 "Directory path "//path//" cannot be deleted")
         end if
 
-      end subroutine check_rmdir
+    end subroutine check_rmdir
 
+    subroutine test_is_absolute_path(error)
+        type(error_t), allocatable, intent(out) :: error
+
+        if (is_absolute_path('.', is_unix=.true.)) then
+            call test_failed(error, "Relative path '.' isn't absolute")
+            return
+        end if
+
+        if (is_absolute_path('abc', is_unix=.true.)) then
+           call test_failed(error, "Relative path 'abc' isn't absolute")
+           return
+        end if
+
+        if (.not. is_absolute_path('/', is_unix=.true.)) then
+            call test_failed(error, "Path '/' is absolute")
+            return
+        end if
+
+        if (.not. is_absolute_path('/abc', is_unix=.true.)) then
+            call test_failed(error, "Path '/abc' is absolute")
+            return
+        end if
+
+        if (.not. is_absolute_path('~/', is_unix=.true.)) then
+            call test_failed(error, "Path '~/' is absolute")
+            return
+        end if
+
+        if (.not. is_absolute_path('~/', is_unix=.true.)) then
+            call test_failed(error, "Path '~/' is absolute")
+            return
+        end if
```

Comment:
Those two check the same condition. What about cases like ``~user/``? Should we really recognize this kind of shell expansion or leave it to the shell to expand those values (passing ``~/`` via the command-line will be expanded by the shell to ``$HOME/`` and we never see the tilde).

Reply:
Yeah, maybe we should not treat `~/` as absolute. 🤔

Reply:
But it's definitely easier to treat it as absolute for now. 😅

Reply:
How would you classify `~\` on Windows? In Powershell this refers to your home, in CMD it does not.

Reply:
We could automatically expand it to the %homepath% if we find it 🤔